codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java

```diff
@@ -144,13 +146,44 @@ protected WebSocketClientHandshaker(URI uri, WebSocketVersion version, String su
     protected WebSocketClientHandshaker(URI uri, WebSocketVersion version, String subprotocol,
                                         HttpHeaders customHeaders, int maxFramePayloadLength,
                                         long forceCloseTimeoutMillis, boolean absoluteUpgradeUrl) {
+        this(uri, version, subprotocol, customHeaders, maxFramePayloadLength, forceCloseTimeoutMillis,
+             false, false);
+    }
+
+    /**
+     * Base constructor
+     *
+     * @param uri
+     *            URL for web socket communications. e.g "ws://myhost.com/mypath". Subsequent web socket frames will be
+     *            sent to this URL.
+     * @param version
+     *            Version of web socket specification to use to connect to the server
+     * @param subprotocol
+     *            Sub protocol request sent to the server.
+     * @param customHeaders
+     *            Map of custom headers to add to the client request
+     * @param maxFramePayloadLength
+     *            Maximum length of a frame's payload
+     * @param forceCloseTimeoutMillis
+     *            Close the connection if it was not closed by the server after timeout specified
+     * @param  absoluteUpgradeUrl
+     *            Use an absolute url for the Upgrade request, typically when connecting through an HTTP proxy over
+     *            clear HTTP
+     * @param  allowClientHandshakeException
+     *             Whether to return a basic {@link WebSocketHandshakeException}
+     *             or client {@link WebSocketClientHandshakeException} that contains a http response
```

Comment:
IMHO, I don't think the extra `allow*HandshakeException` flags are necessary, since both `WebSocketClientHandshakeException` and `WebSocketServerHandshakeException` extend `WebSocketHandshakeException`. I mean, `catch (WebSocketHandshakeException e)` and `e instanceof WebSocketHandshakeException` will still work. I hope no one is using `e.getClass() == WebSocketHandshakeException.class`. I think the extra flags just complicate the code without much benefit.

Reply:
I totally agree about code complication and most likely it not bring a benefit. I hope we can remove it to keep code simple if no one is against.

Reply:
I agree, people should not rely on the concrete exception type such that we cannot throw a sub-class instead.